### PR TITLE
Robot control component

### PR DESCRIPTION
**Pull request by adamdbrw**
A robot control component, first approach:
- Can work with scripts or with a default implementation
- Comes with a default rigid body Twist control
- Comes with an example lua script
- Enables setting of topic and QoS

### CODE REVIEW

File: Code/Source/ROS2ModuleInterface.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include <AzCore/Memory/SystemAllocator.h>
 #include <AzCore/Module/Module.h>
 #include <Frame/ROS2FrameComponent.h>
+#include <RobotControl/ROS2RobotControlComponent.h">
```
Comment (lumberyard-employee-dm) [inline Code/Source/ROS2ModuleInterface.h:11]: There is an additional `"` character in the include path that shouldn't be here

File: Code/Source/RobotControl/ControlConfiguration.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
```
Comment (lumberyard-employee-dm) [inline Code/Source/RobotControl/ControlConfiguration.cpp:9]: The `#pragma once` in a cpp file will cause a compiler warning which will turn into a compile error due to `-Werror`

File: Code/Source/RobotControl/ROS2RobotControlComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
```
Comment (lumberyard-employee-dm) [inline Code/Source/RobotControl/ROS2RobotControlComponent.cpp:8]: Repeat: `#pragma once` needs to be removed

File: Code/Source/RobotControl/RobotConfiguration.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
```
Comment (lumberyard-employee-dm) [inline Code/Source/RobotControl/RobotConfiguration.cpp:9]: repeat: `#pragma once` needs to be removed in order to compile successfully

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

File: Code/Source/RobotControl/RobotControl.h [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                auto ros2Frame = entity->FindComponent<ROS2FrameComponent>();
+                AZStd::string namespacedTopic = ROS2Names::GetNamespacedName(
+                        ros2Frame->GetNamespace(),
+                        m_controlConfiguration.m_topic);
+
+                auto ros2Node = ROS2Interface::Get()->GetNode();
+                m_controlSubscription = ros2Node->create_subscription<T>(
+                    namespacedTopic.data(),
+                    m_controlConfiguration.m_qos.GetQoS(),
+                    std::bind(&RobotControl<T>::OnControlMessage, this, std::placeholders::_1));
```
Comment (lumberyard-employee-dm) [inline Code/Source/RobotControl/RobotControl.h:51]: FYI: Using a lambda results in more [efficient code](https://clang.llvm.org/extra/clang-tidy/checks/modernize-avoid-bind.html) than using std::bind.

```suggestion
                   [this](const T& message) { OnControlMessage(message); });
```

File: Code/Source/ROS2ModuleInterface.h (hunk 1 above)
Comment (adamdbrw) [reply Code/Source/ROS2ModuleInterface.h:11]: My bad - a merge conflict between <> and "" includes was resolved with this rather ugly child!

File: Code/Source/RobotControl/ControlConfiguration.cpp (hunk 2 above)
Comment (adamdbrw) [reply Code/Source/RobotControl/ControlConfiguration.cpp:9]: @pijaro

**Comment by adamdbrw** [comment]
@pijaro for review changes for this PR, please apply them as a commit to develop first, check if everything works and then cherry-pick into this branch.

**Comment by pijaro** [comment]
@lumberyard-employee-dm Thanks for the review :bow:, I applied all of the suggested changes.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**